submissions/yousfiSaad/main.cpp

```diff
@@ -0,0 +1,199 @@
+#include <algorithm>
+#include <cstdio>
+#include <cstring>
+#include <fstream>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+const char *INPUT_FILE_NAME = "input.txt";
+const char *OUTPUT_FILE_NAME = "output.txt";
+const size_t MAX_SIZE = 101;
+
+typedef std::unordered_map<
+    std::string, std::pair<long, std::unordered_map<std::string, long>>>
+    m2;
+
+class FileReader {
+  std::ifstream in{INPUT_FILE_NAME};
+  char reader_buffer[100];
+  bool more = true;
+
+public:
+  bool readLine() {
+    in.getline(reader_buffer, 99);
+    more = !in.eof();
+    return more;
+  }
+  inline char *buffer() { return reader_buffer; }
+  inline bool have_more_lines() { return more; }
+};
+
+long atoi(char *price) {
+  long r = 0;
+  long id;
+  bool hasd = false;
+  int i = 0;
+  for (; price[i] != '\0'; i++) {
+    if (price[i] == '.' || price[i] == ',') {
+      hasd = true;
+      id = i;
+      continue;
+    }
+    r *= 10;
+    r += (price[i] - '0');
+  }
+  if (hasd) {
+    if ((id + 2) == i)
+      r *= 10;
+  } else {
+    r *= 100;
+  }
+  return r;
+}
+
+std::pair<std::string, long> min(m2 &sums) {
+  std::pair<std::string, long> r;
+  r.second = 0;
+  bool first = true;
+  for (m2::iterator it = sums.begin(); it != sums.end(); it++) {
+    if (it->second.first < r.second || first) {
+      first = false;
+      r.first = it->first;
+      r.second = it->second.first;
+    }
+  }
+  return r;
+}
+
+struct cmp {
+  bool operator()(const std::pair<std::string, long> &a,
+                  const std::pair<std::string, long> &b) const {
+    return (a.second > b.second) ||
+           (a.second == b.second && a.first.compare(b.first) > 0);
+  }
+};
+
+void printResult(m2 &sums) {
+  std::pair<std::string, long> r = min(sums);
+
+  FILE *fp = fopen(OUTPUT_FILE_NAME, "w+");
+
+  fprintf(fp, "%s %ld.%02ld\n", r.first.c_str(), r.second / 100,
+          r.second % 100);
+
+  std::vector<std::pair<std::string, long>> v;
+
+  for (const std::pair<const std::string, long> &n : sums[r.first].second)
+    v.push_back(n);
+
+  std::vector<std::pair<std::string, long>>::iterator it = v.begin();
+  for (int i = 0; i < 5 && it != v.end(); ++it, ++i) {
+    std::vector<std::pair<std::string, long>>::iterator min_it =
+        std::max_element(it, v.end(), cmp());
+
+    fprintf(fp, "%s %ld.%02ld\n", min_it->first.c_str(), min_it->second / 100,
```

Comment:
you don't need this

Reply:
Sorry I don't get it.
It's for printing the products